app/controllers/user.js

```diff
@@ -39,6 +43,41 @@ exports.create = (req, res) => {
     });
 };
 
+exports.generate = (req, res) => {
+  const randomPassword = crypto.randomBytes(5).toString('hex');
+  const cardKey = uuidv4();
+  const userObject = {
+    username: req.body.email
```

Comment:
This will connect a user to it's email, isn't it better to just split that to make it impossible to trace it back to a user?

Reply:
We can just make it another random `UUID`. The reason I took the username was to decrease the amount of "random stuff" the user had to type when logging in. But I guess since they will get an email they will `copy-paste` anyway, so might as well give them a random username as well.

One benefit to having a username like this is that we can guarantee people don't get two users. They only have one studmail and VOTE will return the normal "This user already exists" if their studmail is entered twice.

Reply:
> We can just make it another random UUID. The reason I took the username was to decrease the amount of "random stuff" the user had to type when logging in. But I guess since they will get an email they will copy-paste anyway, so might as well give them a random username as well.

:+1: 

> One benefit to having a username like this is that we can guarantee people don't get two users. They only have one studmail and VOTE will return the normal "This user already exists" if their studmail is entered twice.

Yeah, but it kinda defeats the whole purpose of vote then, since we tie a vote user to a real email. Imo we should at least keep them separated, and in a perfect world we would have made this users in an external tool, but that is probably a bit late now (so that no identifiable information ever enters vote.abakus.no).

Also, we have to store the email as a whole, to avoid crash with `odinu@stud.ntnu.no`, `odinu@ntnu.no` and `odinu+bar@gmail.com` (Don't sure if the last one matches, not in the mood to read regexes :sweat_smile:)

Reply:
Yeah, agree, maybe we'll just use a random UUID for the username, then no personal data is linked to the user. But we will then need a process for the election where we write down who has already gotten a user.

Reply:
It would be easiest to just create a new table in mongodb and do more or less the same as we do with `hasVotedUsers`

Reply:
But yeah, if a user claim to "not be able to login", "password not working" or "I got no mail", we have a problem tho :sweat_smile:

Reply:
> It would be easiest to just create a new table in MongoDB and do more or less the same as we do with hasVotedUsers

This is a possibility.